buffer/src/main/java/io/netty/buffer/ByteBufProcessor.java

```diff
@@ -16,10 +16,14 @@
 
 package io.netty.buffer;
 
-public interface ByteBufProcessor {
+/**
+ * Use {@link ByteVisitor} directly where interfaces permit.
+ */
+public interface ByteBufProcessor extends ByteVisitor {
```

Comment:
> Add the overridden methods of the methods who accepts ByteBufProcessor as a parameter, and deprecate the methods with ByteBufProcessor.

I'm not sure we need to have 2 sets of methods.  `ByteBufProcessor extends ByteProcessor`...so anywhere that takes a `ByteProcessor` should be OK to take a `ByteBufProcessor`.  I think just deprecating `ByteBufProcessor` (and public interface within it) and replacing it with `ByteProcessor` would be what we want?